commons/com.b2international.index.tests/src/com/b2international/index/revision/SingleDocumentRevisionIndexTest.java

```diff
@@ -54,6 +54,21 @@ public void updateRevision() throws Exception {
 		indexRevision(MAIN, data);
 		assertEquals(data, getRevision(MAIN, RevisionData.class, STORAGE_KEY1));
 	}
+	
+	@Test
+	public void stageNewRevisionMultipleTimes() throws Exception {
+		final RevisionData data = new RevisionData(STORAGE_KEY1, "fieldA", "fieldB");
+		StagingArea staging = index().prepareCommit(MAIN);
+		staging.stageNew(data, false);
+		
+		staging.stageNew(data, true);
+		assertEquals(1, staging.getNumberOfStagedObjects());
+		assertEquals(data, staging.getNewObject(RevisionData.class, STORAGE_KEY1));
+		
+		staging.stageNew(data, false);
```

Comment:
The implemented test case is a bit different than the one I described, please take a look again.